enhancements/observability/cluster-observability-operator-logging-stack.md

```diff
@@ -0,0 +1,374 @@
+kind: LoggingStack
+metadata:
+  name: sample-logging-stack
+  namespace: openshift-logging
+spec:
+  monitoringSelector:
+    openshift.io/cluster-monitoring: "true"
+  subscription:
+    installPlanApproval: Automatic
+    channel: stable-5.8
+  storage:
+    size: 1x.extra-small
+    storageClassName: gp3-csi
+    storage:
+      secret:
+        name: aws-s3-credentials
+        type: s3
+```
+
+The `ObservabilityUIPlugin` resources are discussed in a [separate proposal](observability-ui-proposal) here.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A.
+
+#### Standalone Clusters
+
+This proposal is mainly targeting standalone clusters either on-premises or OpenShift Dedicated. Users upgrading from existing OpenShift Logging installation need to uninstall them first and re-install through the Cluster Observability Operator by following the above workflow. Doing so will enable to use the correlation component out of the box.
+
+#### Single-node Deployments or MicroShift
+
+N/A.
+
+### Implementation Details/Notes/Constraints
+
+#### Handling multiple LoggingStack resourceselector on the same cluster
+
+Historically the OpenShift Logging product allows the creation **only** of a single logging stack per cluster. This design detail represents a choice to preserve resources on a cluster rather than a limitation. Having multiple end-to-end stacks is rather a very expensive solution and has been corroborated by different means:
+- Soft tenancy combined with full RBAC-based controls on logs. This design choice renders the need for multiple expensive log storage installations obsolete.
+- Using namespaced `ClusterLogForwarder` resources to offers users with configuration options on which workloads should be collected as well as how to filter logs.
+
+Therefore it is possible to allow creating multiple `LoggingStack` resources on the same cluster with the following workflow:
+1. Create main `LoggingStack` resource in `openshift-logging` namespace that reconciles the operator installation as well as the resources `ClusterLogging` and `Lokistack`.
+2. Create a `LoggingStack` resource on any other namespace can be rendered on a `ClusterLogForwarder` resource accounting for collecting the workload logs of that namespace and forwarding them to the `default` log store backed by the `LokiStack` resource.
+
+Limitations of the above worklow are:
+1. Opt-in to collect audit logs should be allowed only via the main `LoggingStack` resource in `openshift-logging`. Any other `LoggingStack` resource will be ignored on this matter.
+2. Using the API audit log filtering capabilities in `ClusterLogForwarder` (See [docs](clf-api-audit-log-filter)) is only possible in `openshift-logging`.
+
+In summary although the above workflow is feasible, the present proposal does **not** recommend any further implementation leaving users to manage their own `ClusterLogForwarder` resource per namespace if needed, because:
```

Comment:
How would COO deal with multiple LoggingStack using different subscriptions anyway? I'm not sure that coupling the subscription with the LoggingStack CRD is the right approach. Shouldn't COO pick up only one LoggingStack object and let users deploy individual ClusterLogForwarders if they need to?

Reply:
> How would COO deal with multiple LoggingStack using different subscriptions anyway? I'm not sure that coupling the subscription with the LoggingStack CRD is the right approach. 

You raise an important aspect here. The proposed `LoggingStack` CRD is highly biased by the fact that OpenShift Logging is designed to be a singelton system. Thus adding the subscription in the CRD looks like an easy task. But I agree with you that we should question if this is the right approach.

To the best of my knowledge CRDs should serve a single purpose like handling the orchestration of a couple of operands that together provide a meaningful system. In comparison my `LoggingStack` CRD serves two things in parallel: a) operators installation (i.e. Cluster Logging Operator, Loki Operator) b) leaf CRD initialization/orchestration (i.e. `ClusterLogging`, `LokiStack`). Is this a bad design? WDYT?

> Shouldn't COO pick up only one LoggingStack object and let users deploy individual ClusterLogForwarders if they need to?

I believe I make this clear above that _"the present proposal does not recommend"_ implemeting ClusterLogForwarder creation via LoggingStack. Simply let users do this per namespace. A LoggingStack per namespace becomes obsolete then.

Reply:
Two thoughts on this:

- We have previously discussed adding a config CRD to COO. This could be place to handle subscription information. Up until now it made more sense to pass operator configurations as cli args to the operator.
- Similarly to a config CRD approach, we could think about making `LoggingStack` a singleton CR.

I agree that adding subscriptions to a CRD that expects multiple CRs feels awkward at best.